### PR TITLE
Rc sync to main

### PR DESCRIPTION
**Pull request by erick-xanadu**

### CODE REVIEW

**Comment by maliasadi** [comment]
@erick-xanadu will you include https://github.com/PennyLaneAI/catalyst/pull/570 before merge?

**Comment by erick-xanadu** [comment]
@maliasadi yeah, but I had two things I wanted to ask / discuss before merging during the meeting today. And it was better to show them separately.

File: frontend/test/pytest/test_template.py [hunk 1]
```diff
@@ -171,6 +171,7 @@ def basis_state_preparation(basis_state):
     assert np.allclose(interpreted_fn(params), jitted_fn(params))
 
 
+@pytest.mark.skip(reason="GlobalPhase not implemented in QIR")
```
Comment (dime10) [inline frontend/test/pytest/test_template.py:174]: This should be removed (my might have missed it in the GP pr?)
```suggestion
@pytest.mark.skip(reason="GlobalPhase not implemented in QIR")
```

**Review by dime10 — CHANGES_REQUESTED** [review]

File: frontend/test/pytest/test_template.py (hunk 1 above)
Comment (erick-xanadu) [reply frontend/test/pytest/test_template.py]: https://github.com/PennyLaneAI/catalyst/pull/571/commits/f434364847fe5b3563d99d2c45b3203415c318e9

File: runtime/Makefile [hunk 2]
```diff
@@ -15,7 +15,7 @@ ENABLE_LIGHTNING_KOKKOS?=ON
 ENABLE_OPENQASM?=ON
 ENABLE_ASAN?=OFF
 # TODO: Update to v0.35.0 after the next lightning release.
-LIGHTNING_GIT_TAG_VALUE?="86f22a768238c3f7c45a6f515b6cb61fc285a636"
+LIGHTNING_GIT_TAG_VALUE?="efb946a736f824835ebf4fd692568d49e6cf1ca5"
```
[the remark this answers is not in the document]
Comment (erick-xanadu) [reply runtime/Makefile]: https://github.com/PennyLaneAI/catalyst/pull/571/commits/f434364847fe5b3563d99d2c45b3203415c318e9

**Comment by rmoyard** [comment]
@erick-xanadu FYI Coverage is not working on the rc branches (similar to PennyLane, we have never found a fix for the issue). Therefore it does not catch missing coverage unitl you want to merge it into master.

**Comment by erick-xanadu** [comment]
@rmoyard are you recommending to improve coverage and then merge? Is it possible to merge and then improve coverage? I believe this PR is only blocked because of @dime10's requesting changes. I just don't want to wait the whole build process again to merge.

**Comment by erick-xanadu** [comment]
@rmoyard Ok! I'll work on it then! Thanks :)

**Review by dime10 — APPROVED** [review]
Sorry about that, this is good to go 👍

**Pull request merged**